osquery/tables/system/linux/deb_packages.cpp

```diff
@@ -142,5 +142,84 @@ QueryData genDebPackages(QueryContext& context) {
     return genDebPackagesImpl(context, logger);
   }
 }
+
+void genDebPackageFiles(RowYield& yield, QueryContext& context) {
+  std::vector<std::string> admindir_list{};
+  std::vector<ErrorLog> errorLogBuffer{};
+
+  if (context.hasConstraint("admindir", EQUALS)) {
+    for (const auto& admindir :
+         context.constraints["admindir"].getAll(EQUALS)) {
+      admindir_list.push_back(admindir);
+    }
+  } else {
+    admindir_list.push_back(kAdminDir);
+  }
+
+  std::set<std::string> allowed_packages;
+  bool has_package_constraint = false;
+  if (context.hasConstraint("package", EQUALS)) {
+    has_package_constraint = true;
+    allowed_packages = context.constraints["package"].getAll(EQUALS);
+  }
+
+  auto dropper = DropPrivileges::get();
+  dropper->dropTo("nobody");
+
+  for (const auto& admindir : admindir_list) {
+    if (!pathExists(admindir).ok()) {
+      continue;
+    }
+    auto dpkg_query_exp = IDpkgQuery::create(admindir);
+    if (dpkg_query_exp.isError()) {
+      errorLogBuffer.emplace_back("Failed to open the dpkg database",
+                                  dpkg_query_exp.takeError(),
+                                  admindir);
+      continue;
+    }
+    auto dpkg_query = dpkg_query_exp.take();
+    auto package_list_exp = dpkg_query->getPackageList();
+    if (package_list_exp.isError()) {
+      errorLogBuffer.emplace_back("Failed to list the packages",
+                                  package_list_exp.takeError(),
+                                  admindir);
+      continue;
+    }
+    auto package_list = package_list_exp.take();
+    for (const auto& package : package_list) {
+      if (has_package_constraint && allowed_packages.count(package.name) == 0) {
```

Comment:
Why have `has_package_constraint` at all?

Reply:
Oh, I guess to differentiate between `IN ()` and unspecified. Okay.

Might be worth a comment

Reply:
I refactored to eliminate this (and hopefully improve performance)